### PR TITLE
rfcs: rnn: bwd: skip weights accumulation

### PR DESCRIPTION
**Pull request by igorsafo**
# Description

A [link](https://github.com/igorsafo/oneDNN/tree/rfcs/rfcs/20221229-rnn-bwd-w-accumulation) to the rendered version.

### CODE REVIEW

**Review by dzarukin — APPROVED** [review]

File: rfcs/20221229-rnn-bwd-w-accumulation/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+```cpp
+// dnnl_types.h
+
+/// Flags for RNN cell.
+typedef enum {
+    /// Undefined RNN flags
+    dnnl_rnn_flags_undef = 0x0,
+    /// Skip weights accumulation
+    dnnl_rnn_flags_skip_weights_accumulation = 0x1
```
Comment (mgouicem) [inline rfcs/20221229-rnn-bwd-w-accumulation/README.md:66]:
```suggestion
    /// Do not add weights gradient to existing diff_weights memory
        dnnl_rnn_flags_diff_weights_overwrite = 0x1
```

File: rfcs/20221229-rnn-bwd-w-accumulation/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+} dnnl_rnn_flags_t;
+
+// dnnl.hpp
+
+/// RNN cell flags.
+enum class rnn_flags : unsigned {
+    /// Undefined RNN flags
+    undef = dnnl_rnn_flags_undef,
+    /// Skip weights accumulation
+    skip_weights_accumulation = dnnl_rnn_flags_skip_weights_accumulation
```
Comment (mgouicem) [inline rfcs/20221229-rnn-bwd-w-accumulation/README.md:76]:
```suggestion
    /// Do not add weights gradient to existing diff_weights memory
        diff_weights_overwrite = 0x1
```

**Review by mgouicem — APPROVED** [review]
LGTM, only thing is about naming, which should explicitly mention diff_weights, and not weights only.

File: rfcs/20221229-rnn-bwd-w-accumulation/README.md (hunk 2 above)
Comment (igorsafo) [reply rfcs/20221229-rnn-bwd-w-accumulation/README.md:76]: @mgouicem , thanks for the variable name. Is there any reason C++ API value is `0x1` and not `dnnl_rnn_flags_diff_weights_overwrite`?

Comment (mgouicem) [reply rfcs/20221229-rnn-bwd-w-accumulation/README.md:76]: My bad, just a copy paste glitch.

**Pull request merged**